crates/cli-support/src/js/mod.rs

```diff
@@ -2116,6 +2116,19 @@ impl<'a> Context<'a> {
         );
     }
 
+    fn expose_assert_char(&mut self) {
+        if !self.should_write_global("assert_char") {
+            return;
+        }
+        self.global(
+            "
+            function _assertChar(c) {
+                if (typeof(c) !== 'number' || c >= 0x110000 || (c >= 0xD800 && c < 0xE000)) throw new Error(`expected a number argument that is a valid Unicode scalar value, found ${c}`);
```

Comment:
'expected a number argument' isn't a great error message here because `c` isn't actually something we received from user code, it's the result of `codePointAt(0)`. Maybe just more explicitly say 'unpaired surrogate found'?

That isn't the right message for the case where it isn't a number though, which can happen if the string is empty. In that case `codePointAt` returns undefined, which gets coerced to `'\0'`. (That means it is technically a change in non-unsound behaviour, but I don't think it really matters.)

Reply:
> That isn't the right message for the case where it isn't a number though, which can happen if the string is empty. In that case `codePointAt` returns undefined, which gets coerced to `'\0'`. (That means it is technically a change in non-unsound behaviour, but I don't think it really matters.)

I addressed this in https://github.com/rustwasm/wasm-bindgen/pull/3866/files#r1509971507, there should be no change in behavior now.